cluster/core/src/multi-jvm/scala/com/lightbend/lagom/internal/cluster/ClusterDistributionSpec.scala

```diff
@@ -25,6 +27,13 @@ class ClusterDistributionSpec extends ClusteredMultiNodeUtils(numOfNodes = 3) wi
     ClusterDistributionSettings(system)
       .copy(ensureActiveInterval = ensureActiveInterval)
 
+  protected override def afterTermination(): Unit = {
+    super.afterTermination()
+
+    enterBarrier("all-nodes-completed-their-assertions")
+    Await.ready(system.terminate(), shutdownTimeout)
```

Comment:
Isn't system.terminate() already handled by the multi-jvm framework?

I'm not familiar with `afterTermination()`, but I get the sense that this will be called once the actor system is terminated, no?

Reply:
yeah, that was wrong, I've reimplemented adding a barrier on a `finally` around the assertion.